src/gc/impl/conservative/gc.d

```diff
@@ -2078,6 +2078,135 @@ struct Gcx
     ToScanStack!(ScanRange) toscanPrecise;
 
     /**
+     * Finds the offset between an index and the next set bit, up to our max.
+     *
+     * Returns: The offset between index and the next bit,
+     *          or -1 if no bits are set between index and max
+     */
+    int bitScanForwardOffset(size_t* bitArray, size_t index, size_t max) nothrow
```

Comment:
>I would not trust the DMD optimizer that recalculation of bitIndex and testBit is efficient

I'm not sure I understand. bitIndex is calculated with just a bit shift and testBit is a bitwise comparison and a bit shift. Aren't those fast operations?

>using floating point log2 is unlikely to be fast

I'm having trouble finding it right now, but I read that the log2 function was a constant time function and that the non floating point overloads just did a cast to `double` themselves. I did find [this](http://stackoverflow.com/a/11398748), which could prove to be an alternative if we need one.

>bsf is undefined for argument 0, this is not Windows specific

When looking at the code for bsf, it appears to me that it always calls softBsf to perform the actual calculations. That is supposed to return -1 when passed 0. It didn't when I happened to be returning -1 on Windows, hence the note. Perhaps I shouldn't have counted on this though.

>bitScanForwardOffset should not be recursive

I agree, and I planned on correcting that once the tests passed.

>core.bitop.bsf(shifted) is called twice

Whoops. One of those was a leftover when I was testing my math.